src/rag/reranker/reranker.py

```diff
@@ -74,21 +75,30 @@ def rerank(
         # filtration to prevent reranking irrelevant documents
         relevant_docs = get_relevant_documents(docs_list, limit=input_limit)
 
-        # reranking using the LLM model
-        response: RerankedDocs = self.chain.invoke(
-            {
-                "documents": format_documents(relevant_docs),
-                "queries": format_queries(queries),
-                "limit": output_limit,
-            }
-        )
-
-        # return reranked documents capped at the output limit
-        reranked_docs = [
-            Document(page_content=doc.page_content)
-            for doc in response.documents[:output_limit]
-        ]
-        return reranked_docs
+        try:
+            # reranking using the LLM model
+            response: RerankedDocs = await run_in_executor(
+                None,
+                self.chain.invoke,
+                {
+                    "documents": format_documents(relevant_docs),
+                    "queries": format_queries(queries),
+                    "limit": output_limit,
+                },
+                None,
+            )
```

Comment:
can you just not use the following instead of run_in_executor?
```
response: RerankedDocs = await self.chain.ainvoke(
            {
                "documents": format_documents(relevant_docs),
                "queries": format_queries(queries),
                "limit": output_limit,
            }
        )
```